angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts

```diff
@@ -196,6 +196,7 @@ export class SidebarNavItemComponent implements AfterContentInit, OnInit, OnDest
 
   onKeyDown(event: KeyboardEvent) {
     let newIndex;
+    // tslint:disable-next-line: deprecation
```

Comment:
Is this a good idea?  I thought we had the linter for a reason

Reply:
I think we are trying to support both keyCode for IE 11 as well as the angular cdk codes for now, I had cases for both.

Reply:
Agreed with Daani on this. @mathsusan after doing some research here's the list of KeyboardEvents (https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent), so while I don't mind the tslint for now, could we add the non-deprecated version .key.

Reply:
so we should talk about the different outputs though (example -'space):
1) .keyCode = 32
2) .code = 'Space' (adds Key or Digit for others)
3) .which = 32
4) .key = ' ' 
5) .charCode = 32

I recommend using .key as default and .charCode as backup.

Reply:
charCode has been the standard for years but it's been deprecated and supposedly being replaced by .key since .code has issues with qwerty